tsctl

```diff
@@ -616,28 +309,136 @@ class SearchTemplateManager(Command):
                 db_session.commit()
 
 
-if __name__ == '__main__':
+class ImportTimeline(Command):
+    """Create a new Timesketch timeline from a file."""
+    option_list = (
+        Option('--file', '-f', dest='file_path', required=True),
+        Option('--sketch_id', '-s', dest='sketch_id', required=False),
+        Option('--username', '-u', dest='username', required=False),
+        Option('--timeline_name', '-n', dest='timeline_name',
+               required=False),
+    )
+
+    def __init__(self):
+        super(ImportTimeline, self).__init__()
+
+    # pylint: disable=arguments-differ, method-hidden
+    def run(self, file_path, sketch_id, username, timeline_name):
+        """This is the run method."""
+
+        file_path = os.path.realpath(file_path)
+        file_path_no_extension, extension = os.path.splitext(file_path)
+        extension = extension.lstrip('.')
+        filename = os.path.basename(file_path_no_extension)
+
+        if not os.path.isfile(file_path):
+            sys.exit('No such file: {0:s}'.format(file_path))
+
+        if extension not in ('plaso', 'csv', 'jsonl'):
+            sys.exit('Unknown extension: {0:s}'.format(file_path))
```

Comment:
can we put the supported extensions in a variable/list and then print them out here... extension FOO is not in the approved list of LIST, or something like that?

Reply:
Done.